src/User/NotificationPreference.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\User;
+
+use Flarum\Database\AbstractModel;
+use Illuminate\Database\Eloquent\Builder;
+use InvalidArgumentException;
+
+/**
+ * @property int $user_id
+ * @property string $type
+ * @property string $channel
+ * @property bool $enabled
+ */
+class NotificationPreference extends AbstractModel
+{
+    protected static $channels = [];
+
+    public function user()
+    {
+        return $this->belongsTo(User::class);
+    }
+
+    public static function addChannel(string $channel, array $defaults = [])
```

Comment:
IMO 'defaults' is a bit ambiguous, maybe something more like 'defaultEnabledTypes'?